presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java

```diff
@@ -291,7 +292,7 @@ public Expression rewriteLogicalBinaryExpression(LogicalBinaryExpression node, N
         public Expression rewriteFunctionCall(FunctionCall node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
         {
             // A grouping() is preceded by a GroupId node when the source node is projecting a groupid symbol
-            if (!source.getOutputSymbols().contains(new Symbol("groupid")) && node.getName().toString().equals(GROUPING)) {
+            if (!source.getOutputSymbols().stream().anyMatch(symbol -> symbol.getName().contains(GROUPID_SYMBOL_HINT)) && node.getName().toString().equals(GROUPING)) {
```

Comment:
https://github.com/Teradata/presto/commit/7fc3a816cd9d005d595b62dbb191e38f5d531699 seems OK, but I wonder if you need grouping changes in `SymbolToInputRewriter`. Without rewrite in `SymbolToInputRewriter` the code (`LocalExecutionPlanner`) would be simpler